crates/control/src/dribble_path_plan.rs

```diff
@@ -0,0 +1,98 @@
+use color_eyre::Result;
+use context_attribute::context;
+use framework::{AdditionalOutput, MainOutput};
+use nalgebra::Point2;
+use spl_network_messages::Team;
+use std::f32::consts::PI;
+use types::{
+    configuration::Behavior, FieldDimensions, GameControllerState, PathObstacle, PathSegment,
+    WorldState,
+};
+
+use crate::behavior::walk_to_pose::WalkPathPlanner;
+
+#[context]
+
+pub struct CreationContext {}
+
+#[context]
+pub struct CycleContext {
+    pub world_state: Input<WorldState, "world_state">,
+    pub field_dimensions: Parameter<FieldDimensions, "field_dimensions">,
+    pub configuration: Parameter<Behavior, "behavior">,
+    pub path_obstacles: AdditionalOutput<Vec<PathObstacle>, "time_to_reach_obstacles">,
+}
+
+#[context]
+#[derive(Default)]
+pub struct MainOutputs {
+    pub dribble_path: MainOutput<Option<Vec<PathSegment>>>,
+}
+
+pub struct DribblePath {}
+impl DribblePath {
+    pub fn new(_context: CreationContext) -> Result<Self> {
+        Ok(Self {})
+    }
+
+    pub fn cycle(&mut self, mut context: CycleContext) -> Result<MainOutputs> {
+        let world_state = context.world_state;
+        let field_dimensions = context.field_dimensions;
+        let configuration = &context.configuration.path_planning;
+        let parameters = &context.configuration.dribbling;
+        let path_obstacles_output = &mut context.path_obstacles;
+        let walk_path_planner =
```

Comment:
maybe add some newlines to visually group together reading parameters, creating path planner, doing sth. with kickdecisions, etc.

Reply:
Done!